### ISSUE TITLE
Decouple the rendering of Joomla\Profiler

### ISSUE THREAD

**Issue by eddieajau** [joomla/joomla-framework#33]
We need to take the rendering elements out of Profiler, or at least decouple it somehow. We should be storing the raw data internally and should be able to return that raw data.

**Comment by dongilbert** [comment]
I wouldn't mind having a rendering layer for the profiler. One great thing about Symfony is its debug / profiler bar. I think we would do well to look at its functionality for some ideas.

**Comment by eddieajau** [comment]
I looked at that too, but I personally don't want to have to deal with media assets "just yet".

**Comment by florianv** [comment]
I think we should store an array of "nodes" in the Profiler 

``` php
$nodes = array(
   'first' => array(
           start_time => xxx,
           start_memory => xxx
    ),

   'second' => array(
           start_time => xxx,
           start_memory => xxx
    ),
);
```

And allow to inject this array in the profiler constructor.

So if we have let's say a renderer like that :

``` php
interface RendererInterface
{
   public function render(Profiler $profiler);
}
```

it is easily testable by injecting a Profiler with test values.

Then in the profiler we could have

``` php
class Profiler
{
   protected $renderer;

   public function render()
   {
     return $this->renderer->render($this);
   }
}
```

**Comment by dongilbert** [comment]
Closed by #74
